python/lsst/alert/database/client/_client.py

```diff
@@ -0,0 +1,175 @@
+# This file is part of alert_database_client.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+import io
+import json
+import gzip
+import struct
+import urllib
+
+import fastavro
+import requests
+
+
+class Client:
+    """
+    A client for the alert database. This client provides access to
+    archived alert packets and their schemas, fetching them over HTTP.
+    """
+
+    def __init__(self, url: str):
+        parsed_url = urllib.parse.urlparse(url)
+        if not parsed_url.scheme:
+            url = "http://" + url
+        self.url = url
+
+        self._schema_cache = {}
+
+    def _get_alert_url(self, alert_id: int) -> str:
+        return urllib.parse.urljoin(self.url, f"/v1/alerts/{alert_id}")
+
+    def get_raw_alert_bytes(self, alert_id: int) -> bytes:
```

Comment:
I wonder if we should consider decorating this with a small `lru_cache`--this would save our servers load if someone (due to testing, or buggy code, etc.) kept requesting the same alert.  Maybe this is premature optimization, but it's easy to implement.

Reply:
Huh, maybe. `lru_cache` is new to me... caches on network requests are never simple. I would hate to cache an error response, for example. What happens if the function decorated with `functools.lru_cache` raises an error?

Reply:
Hmmm, good point.  Suggestion retracted.